lib/cancel-stuck-txs.js

```diff
@@ -0,0 +1,171 @@
+import pRetry from 'p-retry'
+import assert from 'node:assert'
+import * as Sentry from '@sentry/node'
+import ms from 'ms'
+import timers from 'node:timers/promises'
+
+const ROUND_LENGTH_MS = ms('20 minutes')
+
+/**
+  * @param {string} f4addr
+  * @returns 1000 oldest messages
+  */
+async function getMessagesInMempool (f4addr) {
+  const res = await pRetry(
+    () => fetch(`https://filfox.info/api/v1/message/mempool/filtered-list?address=${f4addr}&pageSize=1000`),
+    {
+      async onFailedAttempt (error) {
+        console.warn(error)
+        console.warn('Filfox request failed. Retrying...')
+      }
+    }
+  )
+  if (!res.ok) {
+    throw new Error(`Filfox request failed with ${res.status}: ${(await res.text()).trimEnd()}`)
+  }
+  /** @type {{
+     messages: {
+      cid: string;
+      from: string;
+      to: string;
+      nonce: number;
+      value: string;
+      gasLimit: number;
+      gasFeeCap: string;
+      gasPremium: string;
+      method: string;
+      methodNumber: number;
+      evmMethod: string;
+      createTimestamp: number;
+    }[];
+  }}
+  */
+  const { messages } = /** @type {any} */(await res.json())
+  return messages
+}
+
+/**
+ * @returns {Promise<{
+  "cid": string;
+  "height": number;
+  "timestamp": number;
+  "gasLimit": number;
+  "gasFeeCap": string;
+  "gasPremium": string;
+  "method": string;
+  "methodNumber": number;
+  "receipt": {
+      "exitCode": number;
+      "return": string;
+      "gasUsed": number;
+    },
+    "size": number;
+    "error": string;
+    "baseFee": string;
+    "fee": {
+      "baseFeeBurn": string;
+      "overEstimationBurn": string;
+      "minerPenalty": string;
+      "minerTip": string;
+      "refund": string;
+    },
+  }>}
+*/
+async function getRecentSendMessage () {
+  let res = await fetch('https://filfox.info/api/v1/message/list?method=Send')
```

Comment:
Should we wrap this HTTP request in `pRetry`?

In other words, why do we need to implement retry for the request `https://filfox.info/api/v1/message/mempool/filtered-list` but not for the request for `https://filfox.info/api/v1/message/list`?

Reply:
I added retrying to that other endpoint because I saw it fail, this one I hadn't seen fail. I'm +-0 on adding it, wdyt?

Reply:
I don't have a strong opinion. Adding `pRetry` should be an easy change we can do in the future when the need arises, right?